smac/utils/io/traj_logging.py

```diff
@@ -229,6 +307,32 @@ def _convert_dict_to_config(config_list: typing.List[str], cs: ConfigurationSpac
                 v = float(v)
             elif isinstance(hp, IntegerHyperparameter):
                 v = int(v)
+            elif isinstance(hp, (CategoricalHyperparameter, Constant)):
+                # Checking for the correct type requires jumping some hoops
+                # First, we gather possible interpretations of our string
+                interpretations = [v]
+                if v in ["True", "False"]:
```

Comment:
Wouldn't a boolean be stored as `true` in json representation and then correctly be deserialized?

Reply:
Or is this only for parsing non-json data?

Reply:
That function is only called if we read the old aclib format.